### PR TITLE
Issue 1248

### PR DESCRIPTION
**Pull request by sjspielman**
### Purpose/implementation Section

This PR updates the `mutational-signatures` module to explore the relationship among TP53 status, hypermutators, and signature exposure

#### What GitHub issue does your pull request address?

Issue #1248 

#### Which areas should receive a particularly close look?

Results are now integrated to simultaneously explore signatures 3, 18, and MMR2. There are no strong trends here, but there are some interesting observations for the individual samples that are described in the notebook.

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### What types of results are included (e.g., table, figure)?

Tables and exploratory figures in the notebook. At this time nothing is exported.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: figures/scripts/supp-figureS2.R [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Stephanie J. Spielman for CCDL, 2022
```
Comment (jaclyn-taroni) [inline figures/scripts/supp-figureS2.R:1]: I think this snuck in. Always look at the diff! 😸

File: analyses/mutational-signatures/run_mutational_signatures.sh [hunk 2]
```diff
@@ -49,3 +49,7 @@ Rscript -e "rmarkdown::render('06-compare_cns_exposures.Rmd', clean = TRUE)"
 
 # Visualize results from deconstructSigs fitting
 Rscript -e "rmarkdown::render('07-plot_cns_fit.Rmd', clean = TRUE, params = list(is_ci = ${ABBREVIATED_MUTSIGS}))"
+
+
+# Explore signature relationship with TP53 and hypermutator samples
+Rscript -e "rmarkdown::render('08-explore_hypermutators.Rmd', clean = TRUE)
```
Comment (jaclyn-taroni) [inline analyses/mutational-signatures/run_mutational_signatures.sh:55]:
```suggestion
Rscript -e "rmarkdown::render('08-explore_hypermutators.Rmd', clean = TRUE)

```

**Review by jaclyn-taroni — COMMENTED** [review]
Overall this code/approach looks good! I think you had an unintended file addition here that should get resolved. I also have a question, perhaps for the group, about your hypermutator cutoff choice. Not going to approve, will let @jharenza take a look.

File: analyses/mutational-signatures/08-explore_hypermutators.Rmd [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
++ their exposures to all signatures (with those of interest appearing first: 3, 18, MMR2)
+
+
+
+
+```{r tp53_sigs}
+tmb_sigs_df %>%
+  # filter to the hypermutators
+  dplyr::filter(tmb_type == "coding_tmb", 
+                tmb >= 10,
```
[the remark this answers is not in the document]
Comment (jharenza) [reply analyses/mutational-signatures/08-explore_hypermutators.Rmd:115]: This number was chosen based on the Campbell, et al paper mentioned in #1248, which was a pediatric-specific analysis (10 for hypermutator and 100 for ultra-hypermutator).

File: figures/scripts/supp-figureS2.R (hunk 1 above)
Comment (sjspielman) [reply figures/scripts/supp-figureS2.R:1]: Ahh!! my bad.

**Comment by sjspielman** [comment]
Note the forced push above is because github problems meant the last commit wasn't being sent to CI. Now it's running.

**Review by jharenza — APPROVED** [review]
Thanks for this!

>We do not observe that samples with high MMR2 or signature 3 exposure have high mutational burdens. Moreover, the hypermutators (all are HGG) are unexposed to signature 3 entirely. Notably, three of the hypermutators have >=50% exposure to signature 18.

This is interesting, but another observation is that the ultra-hypermutators (> 100 Mut/Mb) have _some_ MMR signature, whereas the others do not. I did some digging into the pathology reports and appended some manual notes to your table (below). Of note, these four patients do have the genetic predisposition, _Lynch syndrome_, which is defined by defective mismatch repair as a result of germline variants in MLH1, MSH2, MSH6, PMS2, or EPCAM ([ref](https://www.cancer.net/cancer-types/lynch-syndrome)). So that still supports that these patients do have defective MMR even if the signatures weren't "high" as proposed by the initial reference linked here. It is also interesting (strange, or maybe we are onto something new?) that there are a lot of samples with high MMR that are not ultra-hypermutators.

>No samples are exposed to signature 3, 8, or N6
>The HGG hypermutator sample with lowest TMB (TMB=15.48) is primarily exposed to signature 1, which is a normal signature.

We can report these in text.

>>The DMG hypermutator sample (TMB=35.89) is mostly exposed to [signature 11](https://signal.mutationalsignatures.com/explore/referenceCancerSignature/40). This signature was thought to be associated with temozolomide, except that original evidence has not been reproduced.

This tumor (7316-3058) was a recurrence, and when I looked back at the initial tumor (7316-2594), I did see that the patient was treated with temozolomide, so yay, that matches!

| Kids_First_Participant_ID | Kids_First_Biospecimen_ID |        tumor_descriptor | coding_tmb       | TP53_status | Treatment                                                                                                     | cancer_predispositions_pathology                                                              | sample_id |
|---------------------------|---------------------------|-------------------------|------------------|-------------|---------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------|-----------|
| PT_0SPKM4S8               | BS_VW4XN9Y7               | Initial CNS Tumor       | 187.38765        | activated   |        monthly   temodar and CCNU                                                                             | None documented                                                                               | 7316-2640 |
| PT_3CHB9PK5               | BS_20TBZG09               | Initial CNS Tumor       | 307.04922        | loss        | Unknown                                                                                                       | NF-1,Other inherited conditions   NOS                                                         | 7316-515  |
| PT_3CHB9PK5               | BS_8AY2GM4G               | Progressive             |        321.60795 | loss        | Unknown                                                                                                       | NF-1,Other inherited conditions   NOS - constitutional MMR deficiency (Lynch syndrome) - MSH6 | 7316-2085 |
| PT_JNEV57VK               |        BS_85Q5P8GF        | Initial CNS Tumor       | 4.731587         | loss        |        Temozolomide                                                                                           | Other inherited conditions   NOS                                                              | 7316-2594 |
| PT_JNEV57VK               | BS_HM5GFJN8               | Progressive             |        35.89287  | loss        |        Nivolumab   immunotherapy.                                                                             | Other inherited conditions NOS -   Lynch Syndrome, pathogenic MSH2                            | 7316-3058 |
| PT_S0Q27J13               | BS_P3PF53V8               | Initial CNS Tumor       | 15.48265         | activated   |        Temodar,   140mg tablet, once per day for 42 days.      then Beva Irnotecan with temodar every 28 days | None documented                                                                               | 7316-2307 |
| PT_VTM2STE3               | BS_02YBZSBY               | Progressive             | 274.48806        | activated   | Unknown                                                                                                       | Other inherited conditions NOS -   HNPCC/Lynch/turcot syndrome with DMS2 mutation             | 7316-2189 |

I also notice here that "Temodar" treatment for tumors `BS_VW4XN9Y7` and `BS_P3PF53V8` == temozolomide, but they do not have a strong Signature 11. It could be because this was the initial tumor and therapy had not yet had a strong effect on mutation type? I can see if someone can dig up this information so we can discuss in the paper.

Overall, perhaps we can either create a small output table as a main table 2 with some of this information (signatures + clinical + tp53 + treatment). This would entail adding manual treatment and cancer predispositions so that we capture what is in the path report. I can create a separate issue for this. 

Another idea around this was to add perhaps a color gradient to the TP53 score distribution plot based on TMB (or to mark the hyper and ultrahypermutators in a different shape) so that we can tie discussion of this into that figure. In that case, I would suggest some changes to the figures 4 and 5 (also can put into issue):

Fig 3: put panels D,E next to A,B and move C to wide as E 😬 (it's original form). This would get us to discuss mut sigs last, then give a good segue into TP53 scores for figure 4. Then Figure 5 can be the original figure 4, ending with immune scores and potential hazard plots.

Thoughts? We can also have a sidebar!

**Comment by jharenza** [comment]
Updates on the above two for TMZ treatment timelines (looks like tx was after surgery in both cases):
```
7316-2307 - treatment of tem was started 88 days after their biopsy.
7316-2640 - no chemo start and stop dates were recorded but it's noted that radiation started 35 days after surgery. (and from Cassie Kline: "RT always at beginning of therapy regardless.")
```

**Pull request merged**